deweb-index/assembly/contracts/deweb-index.ts

```diff
@@ -0,0 +1,176 @@
+// The entry file of your WebAssembly module.
+import {
+  Address,
+  Context,
+  setBytecode,
+  Storage,
+} from '@massalabs/massa-as-sdk';
+import {
+  Args,
+  bytesToString,
+  i32ToBytes,
+  stringToBytes,
+} from '@massalabs/as-types';
+import {
+  _onlyOwner,
+  _setOwner,
+  OWNER_KEY,
+} from '@massalabs/sc-standards/assembly/contracts/utils/ownership-internal';
+export {
+  setOwner,
+  ownerAddress,
+} from '@massalabs/sc-standards/assembly/contracts/utils/ownership';
+
+export const DEWEB_VERSION_TAG = stringToBytes('\xFFDEWEB_VERSION');
+
+export const INDEX_BY_OWNER_TAG = stringToBytes('\x00INDEX_BY_OWNER');
+
+/**
+ * Initializes the smart contract.
+ * Sets the contract deployer as the owner.
+ */
+export function constructor(_: StaticArray<u8>): void {
+  assert(Context.isDeployingContract());
+
+  _setOwner(Context.caller().toString());
+}
+
+/**
+ * Upgrade the DNS smart contract bytecode
+ * @param args - new bytecode
+ * @returns void
+ */
+export function upgradeSC(args: StaticArray<u8>): void {
+  _onlyOwner();
+
+  setBytecode(args);
+}
+
+/**
+ * Change the owner of the smart contract to the address passed as an argument
+ * @param _binaryArgs - serialized arguments containing the new owner address
+ * @returns void
+ */
+export function transferOwnership(_binaryArgs: StaticArray<u8>): void {
+  _onlyOwner();
+
+  const args = new Args(_binaryArgs);
+  const newOwner = args
+    .nextSerializable<Address>()
+    .expect('Address is missing or invalid');
+
+  _setOwner(newOwner.toString());
+}
+
+/**
+ * Update the website address
+ * @param _binaryArgs - serialized arguments containing the new address
+ * @returns void
+ */
+export function updateWebsite(_binaryArgs: StaticArray<u8>): void {
+  const args = new Args(_binaryArgs);
+  const address = args
+    .nextSerializable<Address>()
+    .expect('Address is missing or invalid');
+
+  _removeWebsite(address);
+
+  if (!Storage.hasOf(address, DEWEB_VERSION_TAG)) return;
+
+  const version = Storage.getOf(address, DEWEB_VERSION_TAG);
+  assert(version.length > 0, 'The address is not a valid DEWEB address');
```

Comment:
remvoe the assertion, also just return because it's an invalid website (so no website)